### ISSUE TITLE
Cilium pod restarts sporadically

### ISSUE THREAD

**Issue by PailaGH** [cilium/cilium#22623]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Sporadically Cilium pod restarts with below error. 

### Cilium Version

1.12.0

### Kernel Version

go1.18.4 linux/amd64

### Kubernetes Version

1.22

### Relevant log output

```shell
level=info msg="Shutting down... " subsys=health-server
level=info msg="Shutting down... " subsys=daemon
level=info msg="Exiting due to signal" signal=terminated subsys=daemon
level=info msg="HTTP server Shutdown: context deadline exceeded" subsys=health-server
level=info msg="Stopped serving cilium API at unix:///var/run/cilium/cilium.sock" subsys=daemon
level=info msg="Stopped serving cilium health API at unix:///var/run/cilium/health.sock" subsys=health-server
level=info msg="Stopping fswatcher" config=tls-server subsys=hubble
level=info msg="HTTP server Shutdown: context deadline exceeded" subsys=daemon
level=info msg="Waiting for all endpoints' go routines to be stopped." subsys=daemon
level=info msg="Stopped reading perf buffer" startTime="2022-12-04 09:34:27.209761593 +0000 UTC m=+3.043854206" subsys=monitor-agent


Can you please help?
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by PailaGH** [comment]
Cilium pod start with following logs:
level=info msg="Started gops server" address="127.0.0.1:9890" subsys=daemon
level=info msg="Memory available for map entries (0.003% of 32834203648B): 82085509B" subsys=config
level=info msg="option bpf-ct-global-tcp-max set by dynamic sizing to 288019" subsys=config
level=info msg="option bpf-ct-global-any-max set by dynamic sizing to 144009" subsys=config
level=info msg="option bpf-nat-global-max set by dynamic sizing to 288019" subsys=config
level=info msg="option bpf-neigh-global-max set by dynamic sizing to 288019" subsys=config
level=info msg="option bpf-sock-rev-map-max set by dynamic sizing to 144009" subsys=config
level=info msg="  --agent-health-port='9879'" subsys=daemon
level=info msg="  --agent-labels=''" subsys=daemon
level=info msg="  --agent-not-ready-taint-key='node.cilium.io/agent-not-ready'" subsys=daemon
level=info msg="  --allocator-list-timeout='3m0s'" subsys=daemon
level=info msg="  --allow-icmp-frag-needed='true'" subsys=daemon
level=info msg="  --allow-localhost='auto'" subsys=daemon
level=info msg="  --annotate-k8s-node='false'" subsys=daemon
level=info msg="  --api-rate-limit=''" subsys=daemon
level=info msg="  --arping-refresh-period='30s'" subsys=daemon
level=info msg="  --auto-create-cilium-node-resource='true'" subsys=daemon
level=info msg="  --auto-direct-node-routes='false'" subsys=daemon
level=info msg="  --bgp-announce-lb-ip='false'" subsys=daemon
level=info msg="  --bgp-announce-pod-cidr='false'" subsys=daemon
level=info msg="  --bgp-config-path='/var/lib/cilium/bgp/config.yaml'" subsys=daemon
level=info msg="  --bpf-ct-global-any-max='262144'" subsys=daemon
level=info msg="  --bpf-ct-global-tcp-max='524288'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-fin='10s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-syn='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp-grace='1m0s'" subsys=daemon
level=info msg="  --bpf-filter-priority='1'" subsys=daemon
level=info msg="  --bpf-fragments-map-max='8192'" subsys=daemon
level=info msg="  --bpf-lb-acceleration='disabled'" subsys=daemon
level=info msg="  --bpf-lb-affinity-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-algorithm='random'" subsys=daemon
level=info msg="  --bpf-lb-dev-ip-addr-inherit=''" subsys=daemon
level=info msg="  --bpf-lb-dsr-dispatch='opt'" subsys=daemon
level=info msg="  --bpf-lb-dsr-l4-xlate='frontend'" subsys=daemon
level=info msg="  --bpf-lb-external-clusterip='false'" subsys=daemon
level=info msg="  --bpf-lb-maglev-hash-seed='JLfvgnHc2kaSUFaI'" subsys=daemon
level=info msg="  --bpf-lb-maglev-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-maglev-table-size='16381'" subsys=daemon
level=info msg="  --bpf-lb-map-max='65536'" subsys=daemon
level=info msg="  --bpf-lb-mode='snat'" subsys=daemon
level=info msg="  --bpf-lb-rev-nat-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv4-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv6-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-service-backend-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-service-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-sock='false'" subsys=daemon
level=info msg="  --bpf-lb-sock-hostns-only='false'" subsys=daemon
level=info msg="  --bpf-lb-source-range-map-max='0'" subsys=daemon
level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon
level=info msg="  --bpf-nat-global-max='524288'" subsys=daemon
level=info msg="  --bpf-neigh-global-max='524288'" subsys=daemon
level=info msg="  --bpf-policy-map-max='16384'" subsys=daemon
level=info msg="  --bpf-root='/sys/fs/bpf'" subsys=daemon
level=info msg="  --bpf-sock-rev-map-max='262144'" subsys=daemon
level=info msg="  --bypass-ip-availability-upon-restore='false'" subsys=daemon
level=info msg="  --certificates-directory='/var/run/cilium/certs'" subsys=daemon
level=info msg="  --cflags=''" subsys=daemon
level=info msg="  --cgroup-root='/run/cilium/cgroupv2'" subsys=daemon
level=info msg="  --cluster-health-port='4240'" subsys=daemon
level=info msg="  --cluster-id='0'" subsys=daemon
level=info msg="  --cluster-name='default'" subsys=daemon
level=info msg="  --clustermesh-config='/var/lib/cilium/clustermesh/'" subsys=daemon
level=info msg="  --cmdref=''" subsys=daemon
level=info msg="  --config=''" subsys=daemon
level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=daemon
level=info msg="  --conntrack-gc-interval='0s'" subsys=daemon
level=info msg="  --crd-wait-timeout='5m0s'" subsys=daemon
level=info msg="  --datapath-mode='veth'" subsys=daemon
level=info msg="  --debug='false'" subsys=daemon
level=info msg="  --debug-verbose=''" subsys=daemon
level=info msg="  --derive-masquerade-ip-addr-from-device=''" subsys=daemon
level=info msg="  --devices=''" subsys=daemon
level=info msg="  --direct-routing-device=''" subsys=daemon
level=info msg="  --disable-cnp-status-updates='true'" subsys=daemon
level=info msg="  --disable-conntrack='false'" subsys=daemon
level=info msg="  --disable-endpoint-crd='false'" subsys=daemon
level=info msg="  --disable-envoy-version-check='false'" subsys=daemon
level=info msg="  --disable-iptables-feeder-rules=''" subsys=daemon
level=info msg="  --dns-max-ips-per-restored-rule='1000'" subsys=daemon
level=info msg="  --dns-policy-unload-on-shutdown='false'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-limit='0'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-processing-grace-period='0s'" subsys=daemon
level=info msg="  --egress-masquerade-interfaces=''" subsys=daemon
level=info msg="  --egress-multi-home-ip-rule-compat='false'" subsys=daemon
level=info msg="  --enable-auto-protect-node-port-range='true'" subsys=daemon
level=info msg="  --enable-bandwidth-manager='false'" subsys=daemon
level=info msg="  --enable-bbr='false'" subsys=daemon
level=info msg="  --enable-bgp-control-plane='false'" subsys=daemon
level=info msg="  --enable-bpf-clock-probe='true'" subsys=daemon
level=info msg="  --enable-bpf-masquerade='false'" subsys=daemon
level=info msg="  --enable-bpf-tproxy='false'" subsys=daemon
level=info msg="  --enable-cilium-endpoint-slice='false'" subsys=daemon
level=info msg="  --enable-custom-calls='false'" subsys=daemon
level=info msg="  --enable-endpoint-health-checking='false'" subsys=daemon
level=info msg="  --enable-endpoint-routes='false'" subsys=daemon
level=info msg="  --enable-envoy-config='false'" subsys=daemon
level=info msg="  --enable-external-ips='true'" subsys=daemon
level=info msg="  --enable-health-check-nodeport='true'" subsys=daemon
level=info msg="  --enable-health-checking='true'" subsys=daemon
level=info msg="  --enable-host-firewall='false'" subsys=daemon
level=info msg="  --enable-host-legacy-routing='false'" subsys=daemon
level=info msg="  --enable-host-port='true'" subsys=daemon
level=info msg="  --enable-host-reachable-services='false'" subsys=daemon
level=info msg="  --enable-hubble='true'" subsys=daemon
level=info msg="  --enable-hubble-recorder-api='true'" subsys=daemon
level=info msg="  --enable-icmp-rules='true'" subsys=daemon
level=info msg="  --enable-identity-mark='true'" subsys=daemon
level=info msg="  --enable-ip-masq-agent='false'" subsys=daemon
level=info msg="  --enable-ipsec='false'" subsys=daemon
level=info msg="  --enable-ipv4='true'" subsys=daemon
level=info msg="  --enable-ipv4-egress-gateway='false'" subsys=daemon
level=info msg="  --enable-ipv4-fragment-tracking='true'" subsys=daemon
level=info msg="  --enable-ipv4-masquerade='false'" subsys=daemon
level=info msg="  --enable-ipv6='false'" subsys=daemon
level=info msg="  --enable-ipv6-masquerade='false'" subsys=daemon
level=info msg="  --enable-ipv6-ndp='false'" subsys=daemon
level=info msg="  --enable-k8s-api-discovery='false'" subsys=daemon
level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=daemon
level=info msg="  --enable-k8s-event-handover='false'" subsys=daemon
level=info msg="  --enable-k8s-terminating-endpoint='true'" subsys=daemon
level=info msg="  --enable-l2-neigh-discovery='true'" subsys=daemon
level=info msg="  --enable-l7-proxy='true'" subsys=daemon
level=info msg="  --enable-local-node-route='false'" subsys=daemon
level=info msg="  --enable-local-redirect-policy='false'" subsys=daemon
level=info msg="  --enable-mke='false'" subsys=daemon
level=info msg="  --enable-monitor='true'" subsys=daemon
level=info msg="  --enable-node-port='false'" subsys=daemon
level=info msg="  --enable-policy='default'" subsys=daemon
level=info msg="  --enable-recorder='false'" subsys=daemon
level=info msg="  --enable-remote-node-identity='true'" subsys=daemon
level=info msg="  --enable-runtime-device-detection='false'" subsys=daemon
level=info msg="  --enable-selective-regeneration='true'" subsys=daemon
level=info msg="  --enable-service-topology='false'" subsys=daemon
level=info msg="  --enable-session-affinity='false'" subsys=daemon
level=info msg="  --enable-svc-source-range-check='true'" subsys=daemon
level=info msg="  --enable-tracing='false'" subsys=daemon
level=info msg="  --enable-unreachable-routes='false'" subsys=daemon
level=info msg="  --enable-vtep='false'" subsys=daemon
level=info msg="  --enable-well-known-identities='false'" subsys=daemon
level=info msg="  --enable-wireguard='false'" subsys=daemon
level=info msg="  --enable-wireguard-userspace-fallback='false'" subsys=daemon
level=info msg="  --enable-xdp-prefilter='false'" subsys=daemon
level=info msg="  --enable-xt-socket-fallback='true'" subsys=daemon
level=info msg="  --encrypt-interface=''" subsys=daemon
level=info msg="  --encrypt-node='false'" subsys=daemon
level=info msg="  --endpoint-gc-interval='5m0s'" subsys=daemon
level=info msg="  --endpoint-interface-name-prefix=''" subsys=daemon
level=info msg="  --endpoint-queue-size='25'" subsys=daemon
level=info msg="  --endpoint-status=''" subsys=daemon
level=info msg="  --envoy-config-timeout='2m0s'" subsys=daemon
level=info msg="  --envoy-log=''" subsys=daemon
level=info msg="  --exclude-local-address=''" subsys=daemon
level=info msg="  --fixed-identity-mapping=''" subsys=daemon
level=info msg="  --force-local-policy-eval-at-source='true'" subsys=daemon
level=info msg="  --fqdn-regex-compile-lru-size='1024'" subsys=daemon
level=info msg="  --gops-port='9890'" subsys=daemon
level=info msg="  --host-reachable-services-protos='tcp,udp'" subsys=daemon
level=info msg="  --http-403-msg=''" subsys=daemon
level=info msg="  --http-idle-timeout='0'" subsys=daemon
level=info msg="  --http-max-grpc-timeout='0'" subsys=daemon
level=info msg="  --http-normalize-path='true'" subsys=daemon
level=info msg="  --http-request-timeout='3600'" subsys=daemon
level=info msg="  --http-retry-count='3'" subsys=daemon
level=info msg="  --http-retry-timeout='0'" subsys=daemon
level=info msg="  --hubble-disable-tls='false'" subsys=daemon
level=info msg="  --hubble-event-buffer-capacity='4095'" subsys=daemon
level=info msg="  --hubble-event-queue-size='0'" subsys=daemon
level=info msg="  --hubble-export-file-compress='false'" subsys=daemon
level=info msg="  --hubble-export-file-max-backups='5'" subsys=daemon
level=info msg="  --hubble-export-file-max-size-mb='10'" subsys=daemon
level=info msg="  --hubble-export-file-path=''" subsys=daemon
level=info msg="  --hubble-listen-address=':4244'" subsys=daemon
level=info msg="  --hubble-metrics=''" subsys=daemon
level=info msg="  --hubble-metrics-server=''" subsys=daemon
level=info msg="  --hubble-recorder-sink-queue-size='1024'" subsys=daemon
level=info msg="  --hubble-recorder-storage-path='/var/run/cilium/pcaps'" subsys=daemon
level=info msg="  --hubble-socket-path='/var/run/cilium/hubble.sock'" subsys=daemon
level=info msg="  --hubble-tls-cert-file='/var/lib/cilium/tls/hubble/server.crt'" subsys=daemon
level=info msg="  --hubble-tls-client-ca-files='/var/lib/cilium/tls/hubble/client-ca.crt'" subsys=daemon
level=info msg="  --hubble-tls-key-file='/var/lib/cilium/tls/hubble/server.key'" subsys=daemon
level=info msg="  --identity-allocation-mode='crd'" subsys=daemon
level=info msg="  --identity-change-grace-period='5s'" subsys=daemon
level=info msg="  --identity-restore-grace-period='10m0s'" subsys=daemon
level=info msg="  --install-egress-gateway-routes='false'" subsys=daemon
level=info msg="  --install-iptables-rules='true'" subsys=daemon
level=info msg="  --install-no-conntrack-iptables-rules='false'" subsys=daemon
level=info msg="  --ip-allocation-timeout='2m0s'" subsys=daemon
level=info msg="  --ip-masq-agent-config-path='/etc/config/ip-masq-agent'" subsys=daemon
level=info msg="  --ipam='cluster-pool'" subsys=daemon
level=info msg="  --ipsec-key-file=''" subsys=daemon
level=info msg="  --iptables-lock-timeout='5s'" subsys=daemon
level=info msg="  --iptables-random-fully='false'" subsys=daemon
level=info msg="  --ipv4-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv4-node='auto'" subsys=daemon
level=info msg="  --ipv4-pod-subnets=''" subsys=daemon
level=info msg="  --ipv4-range='auto'" subsys=daemon
level=info msg="  --ipv4-service-loopback-address='169.254.42.1'" subsys=daemon
level=info msg="  --ipv4-service-range='auto'" subsys=daemon
level=info msg="  --ipv6-cluster-alloc-cidr='f00d::/64'" subsys=daemon
level=info msg="  --ipv6-mcast-device=''" subsys=daemon
level=info msg="  --ipv6-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv6-node='auto'" subsys=daemon
level=info msg="  --ipv6-pod-subnets=''" subsys=daemon
level=info msg="  --ipv6-range='auto'" subsys=daemon
level=info msg="  --ipv6-service-range='auto'" subsys=daemon
level=info msg="  --ipvlan-master-device='undefined'" subsys=daemon
level=info msg="  --join-cluster='false'" subsys=daemon
level=info msg="  --k8s-api-server=''" subsys=daemon
level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=daemon
level=info msg="  --k8s-kubeconfig-path=''" subsys=daemon
level=info msg="  --k8s-namespace='kube-system'" subsys=daemon
level=info msg="  --k8s-require-ipv4-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-require-ipv6-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-service-cache-size='128'" subsys=daemon
level=info msg="  --k8s-service-proxy-name=''" subsys=daemon
level=info msg="  --k8s-sync-timeout='3m0s'" subsys=daemon
level=info msg="  --k8s-watcher-endpoint-selector='metadata.name!=kube-scheduler,metadata.name!=kube-controller-manager,metadata.name!=etcd-operator,metadata.name!=gcp-controller-manager'" subsys=daemon
level=info msg="  --keep-config='false'" subsys=daemon
level=info msg="  --kube-proxy-replacement='disabled'" subsys=daemon
level=info msg="  --kube-proxy-replacement-healthz-bind-address=''" subsys=daemon
level=info msg="  --kvstore=''" subsys=daemon
level=info msg="  --kvstore-connectivity-timeout='2m0s'" subsys=daemon
level=info msg="  --kvstore-lease-ttl='15m0s'" subsys=daemon
level=info msg="  --kvstore-max-consecutive-quorum-errors='2'" subsys=daemon
level=info msg="  --kvstore-opt=''" subsys=daemon
level=info msg="  --kvstore-periodic-sync='5m0s'" subsys=daemon
level=info msg="  --label-prefix-file=''" subsys=daemon
level=info msg="  --labels=''" subsys=daemon
level=info msg="  --lib-dir='/var/lib/cilium'" subsys=daemon
level=info msg="  --local-max-addr-scope='252'" subsys=daemon
level=info msg="  --local-router-ipv4=''" subsys=daemon
level=info msg="  --local-router-ipv6=''" subsys=daemon
level=info msg="  --log-driver=''" subsys=daemon
level=info msg="  --log-opt=''" subsys=daemon
level=info msg="  --log-system-load='false'" subsys=daemon
level=info msg="  --max-controller-interval='0'" subsys=daemon
level=info msg="  --metrics=''" subsys=daemon
level=info msg="  --mke-cgroup-mount=''" subsys=daemon
level=info msg="  --monitor-aggregation='medium'" subsys=daemon
level=info msg="  --monitor-aggregation-flags='all'" subsys=daemon
level=info msg="  --monitor-aggregation-interval='5s'" subsys=daemon
level=info msg="  --monitor-queue-size='0'" subsys=daemon
level=info msg="  --mtu='0'" subsys=daemon
level=info msg="  --node-port-acceleration='disabled'" subsys=daemon
level=info msg="  --node-port-algorithm='random'" subsys=daemon
level=info msg="  --node-port-bind-protection='true'" subsys=daemon
level=info msg="  --node-port-mode='snat'" subsys=daemon
level=info msg="  --node-port-range='30000,32767'" subsys=daemon
level=info msg="  --policy-audit-mode='false'" subsys=daemon
level=info msg="  --policy-queue-size='100'" subsys=daemon
level=info msg="  --policy-trigger-interval='1s'" subsys=daemon
level=info msg="  --pprof='false'" subsys=daemon
level=info msg="  --pprof-port='6060'" subsys=daemon
level=info msg="  --preallocate-bpf-maps='false'" subsys=daemon
level=info msg="  --prepend-iptables-chains='true'" subsys=daemon
level=info msg="  --procfs='/host/proc'" subsys=daemon
level=info msg="  --prometheus-serve-addr=':9962'" subsys=daemon
level=info msg="  --proxy-connect-timeout='1'" subsys=daemon
level=info msg="  --proxy-gid='1337'" subsys=daemon
level=info msg="  --proxy-max-connection-duration-seconds='0'" subsys=daemon
level=info msg="  --proxy-max-requests-per-connection='0'" subsys=daemon
level=info msg="  --proxy-prometheus-port='0'" subsys=daemon
level=info msg="  --read-cni-conf=''" subsys=daemon
level=info msg="  --restore='true'" subsys=daemon
level=info msg="  --route-metric='0'" subsys=daemon
level=info msg="  --sidecar-istio-proxy-image='cilium/istio_proxy'" subsys=daemon
level=info msg="  --single-cluster-route='false'" subsys=daemon
level=info msg="  --socket-path='/var/run/cilium/cilium.sock'" subsys=daemon
level=info msg="  --sockops-enable='false'" subsys=daemon
level=info msg="  --state-dir='/var/run/cilium'" subsys=daemon
level=info msg="  --tofqdns-dns-reject-response-code='refused'" subsys=daemon
level=info msg="  --tofqdns-enable-dns-compression='true'" subsys=daemon
level=info msg="  --tofqdns-endpoint-max-ip-per-hostname='50'" subsys=daemon
level=info msg="  --tofqdns-idle-connection-grace-period='0s'" subsys=daemon
level=info msg="  --tofqdns-max-deferred-connection-deletes='10000'" subsys=daemon
level=info msg="  --tofqdns-min-ttl='3600'" subsys=daemon
level=info msg="  --tofqdns-pre-cache=''" subsys=daemon
level=info msg="  --tofqdns-proxy-port='0'" subsys=daemon
level=info msg="  --tofqdns-proxy-response-max-delay='100ms'" subsys=daemon
level=info msg="  --trace-payloadlen='128'" subsys=daemon
level=info msg="  --tunnel='disabled'" subsys=daemon
level=info msg="  --tunnel-port='0'" subsys=daemon
level=info msg="  --version='false'" subsys=daemon
level=info msg="  --vlan-bpf-bypass=''" subsys=daemon
level=info msg="  --vtep-cidr=''" subsys=daemon
level=info msg="  --vtep-endpoint=''" subsys=daemon
level=info msg="  --vtep-mac=''" subsys=daemon
level=info msg="  --vtep-mask=''" subsys=daemon
level=info msg="  --write-cni-conf-when-ready=''" subsys=daemon
level=info msg="     _ _ _" subsys=daemon
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
level=info msg="|  _| | | | | |     |" subsys=daemon
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
level=info msg="Cilium 1.12.0 9447cd1 2022-07-19T12:22:00+02:00 go version go1.18.4 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: e90612180b82d07c124bbf8e1ffe94a8d603f8ae/1.21.3/Distribution/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (5.4.209) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
level=info msg=" - reserved:.*" subsys=labels-filter
level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:k8s\\.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Auto-disabling \"enable-bpf-clock-probe\" feature since KERNEL_HZ cannot be determined" error="Cannot probe CONFIG_HZ" subsys=daemon
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.62.0.0/16
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Establishing connection to apiserver" host="https://172.20.0.1:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=info msg="Auto-disabling \"enable-node-port\", \"enable-external-ips\", \"bpf-lb-sock\", \"enable-host-port\" features and falling back to \"enable-host-legacy-routing\"" subsys=daemon
level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=10.134.0.62 mtu=9001 subsys=mtu
level=info msg="Restoring 9 old CIDR identities" subsys=daemon
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=info msg="Restored backends from maps" failedBackends=0 restoredBackends=95 subsys=service
level=info msg="Restored services from maps" failedServices=0 restoredServices=43 subsys=service
level=info msg="Reading old endpoints..." subsys=daemon
level=warning msg="Found incomplete restore directory /var/run/cilium/state/1009_next. Removing it..." endpointID=1009_next subsys=endpoint
level=warning msg="Found incomplete restore directory /var/run/cilium/state/313_next. Removing it..." endpointID=313_next subsys=endpoint
level=warning msg="Found incomplete restore directory /var/run/cilium/state/1647_next. Removing it..." endpointID=1647_next subsys=endpoint
level=warning msg="Found incomplete restore directory /var/run/cilium/state/1040_next. Removing it..." endpointID=1040_next subsys=endpoint
level=warning msg="Found incomplete restore directory /var/run/cilium/state/225_next. Removing it..." endpointID=225_next subsys=endpoint
level=warning msg="Evicting expired DNS cache entries that may be in-use due to per-host limits. This may cause recently created connections to be disconnected. Raise tofqdns-endpoint-max-ip-per-hostname to mitigate this." subsys=fqdn
level=warning msg="Evicting expired DNS cache entries that may be in-use due to per-host limits. This may cause recently created connections to be disconnected. Raise tofqdns-endpoint-max-ip-per-hostname to mitigate this." subsys=fqdn
level=info msg="Reusing previous DNS proxy port: 33973" subsys=daemon
level=info msg="Waiting until all Cilium CRDs are available" subsys=k8s
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="All Cilium CRDs have been found and are available" subsys=k8s
level=info msg="Creating or updating CiliumNode resource" node=ip-10-134-0-62.ec2.internal subsys=nodediscovery
level=info msg="Retrieved node information from cilium node" nodeName=ip-10-134-0-62.ec2.internal subsys=k8s
level=info msg="Received own node information from API server" ipAddr.ipv4=10.134.0.62 ipAddr.ipv6="<nil>" k8sNodeIP=10.134.0.62 labels="map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/instance-type:m5.2xlarge beta.kubernetes.io/os:linux failure-domain.beta.kubernetes.io/region:us-east-1 failure-domain.beta.kubernetes.io/zone:us-east-1a k8s.io/cloud-provider-aws:4e91b393e7a86bae604ee7880400034a kubernetes.io/arch:amd64 kubernetes.io/hostname:ip-10-134-0-62.ec2.internal kubernetes.io/os:linux node.kubernetes.io/instance-type:m5.2xlarge topology.kubernetes.io/region:us-east-1 topology.kubernetes.io/zone:us-east-1a]" nodeName=ip-10-134-0-62.ec2.internal subsys=k8s v4Prefix=10.0.5.0/24 v6Prefix="<nil>"
level=info msg="k8s mode: Allowing localhost to reach local endpoints" subsys=daemon
level=info msg="Detected devices" devices="[]" subsys=linux-datapath
level=info msg="Enabling k8s event listener" subsys=k8s-watcher
level=info msg="Removing stale endpoint interfaces" subsys=daemon
level=info msg="Skipping kvstore configuration" subsys=daemon
level=info msg="Waiting until local node addressing before starting watchers depending on it" subsys=k8s-watcher
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"insights-admin\",\"k8s:io.kubernetes.pod.namespace\":\"ns-produse1\"}} NodeSelector:{} Ingress:[{IngressCommonRule:{FromEndpoints:[{\"matchLabels\":{\"k8s:app\":\"k8s-ingress\",\"k8s:io.kubernetes.pod.namespace\":\"ns-produse1\"}}] FromRequires:[] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} ToPorts:[{Ports:[{Port:8080 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ICMPs:[]}] IngressDeny:[] Egress:[] EgressDeny:[] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=insights-admin-ingress-network-policy k8s:io.cilium.k8s.policy.namespace=ns-produse1 k8s:io.cilium.k8s.policy.uid=4d39d7c6-d2b9-4719-b3be-8d253c377261] Description:}]" policyAddRequest=4ab53dcc-082c-4603-bfe3-568e6b89167d subsys=daemon
level=info msg="Restored router address from node_config" file=/var/run/cilium/state/globals/node_config.h ipv4=10.0.5.233 ipv6="<nil>" subsys=node
level=info msg="Policy imported via API, recalculating..." policyAddRequest=4ab53dcc-082c-4603-bfe3-568e6b89167d policyRevision=2 subsys=daemon
level=info msg="Initializing node addressing" subsys=daemon
level=info msg="NetworkPolicy successfully added" k8sApiVersion= k8sNetworkPolicyName=insights-admin-ingress-network-policy subsys=k8s-watcher
level=info msg="Initializing cluster-pool IPAM" subsys=ipam v4Prefix=10.0.5.0/24 v6Prefix="<nil>"
level=info msg="Restoring endpoints..." subsys=daemon
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"csp-internal\",\"k8s:io.kubernetes.pod.namespace\":\"ns-produse1\"}} NodeSelector:{} Ingress:[{IngressCommonRule:{FromEndpoints:[{\"matchLabels\":{\"k8s:app\":\"k8s-ingress\",\"k8s:io.kubernetes.pod.namespace\":\"ns-produse1\"}}] FromRequires:[] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} ToPorts:[{Ports:[{Port:8443 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>} {Ports:[{Port:443 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ICMPs:[]}] IngressDeny:[] Egress:[] EgressDeny:[] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=csp-internal-ingress-network-policy k8s:io.cilium.k8s.policy.namespace=ns-produse1 k8s:io.cilium.k8s.policy.uid=0b25f971-ce83-4654-955c-5d993cba7e5f] Description:}]" policyAddRequest=1bb5bd43-86fc-406f-a523-cdbdd79c1899 subsys=daemon
level=info msg="Policy imported via API, recalculating..." policyAddRequest=1bb5bd43-86fc-406f-a523-cdbdd79c1899 policyRevision=3 subsys=daemon
level=info msg="NetworkPolicy successfully added" k8sApiVersion= k8sNetworkPolicyName=csp-internal-ingress-network-policy subsys=k8s-watcher
level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"dms-fts-initial-index-broker\",\"k8s:io.kubernetes.pod.namespace\":\"ns-produse1\"}} NodeSelector:{} Ingress:[{IngressCommonRule:{FromEndpoints:[{\"matchLabels\":{\"k8s:app\":\"k8s-ingress\",\"k8s:io.kubernetes.pod.namespace\":\"ns-produse1\"}}] FromRequires:[] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} ToPorts:[{Ports:[{Port:8080 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ICMPs:[]}] IngressDeny:[] Egress:[] EgressDeny:[] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=dms-fts-initial-index-broker-ingress-network-policy k8s:io.cilium.k8s.policy.namespace=ns-produse1 k8s:io.cilium.k8s.policy.uid=58280d05-967e-457f-87de-2da9fb5ac3a4] Description:}]" policyAddRequest=6e9013ea-9ba2-44ea-a35f-8bc5595f7744 subsys=daemon

**Comment by PailaGH** [comment]
another noticed is cilium pod is terminating with error code: 137
which seems to be out of memory.

**Comment by pchaigno** [comment]
- Could you share a sysdump of the cluster?
- Given your previous comment, did you check memory usage on the node at the time of the Cilium agent restart?

**Comment by PailaGH** [comment]
Hello Paul
Please see attached sysdump file.
 cilium-sysdump-20221213-094907.zip
<https://drive.google.com/file/d/1HFBdyrYevDjQ09jUP-ZTUxl4YvgNkjyB/view?usp=drive_web>

Thanks
Paila

**Comment by PailaGH** [comment]
[cilium-sysdump-20221213-094907.zip](https://github.com/cilium/cilium/files/10224042/cilium-sysdump-20221213-094907.zip)

**Comment by pchaigno** [comment]
The node where the agent restarted seems a bit low on free memory so that could have been the issue. You'll need to properly collect your resource usage over time to confirm.

**Comment by PailaGH** [comment]
we observed that Cilium deployment does not have resource request and limits defined. we are trying to set the limits and monitor the behavior.
btw, which log/parameter did you see to find the node is low on free memory?

**Comment by pchaigno** [comment]
I simply checked the top output in the bugtool report.

> we observed that Cilium deployment does not have resource request and limits defined. we are trying to set the limits and monitor the behavior.

I see. So this isn't actually a bug but an expected OOM, right? Let's close this issue then.

**Comment by PailaGH** [comment]
Is it recommended to define resource limits of Cilium pod?

**Comment by PailaGH** [comment]
![MemoryUsage_workernode](https://user-images.githubusercontent.com/25868810/207889734-fcedec5e-dfa4-4861-b674-f64b8f6f9085.png)
worker node memory usage around restart time. 
![MemoryUsage_CiliumPod](https://user-images.githubusercontent.com/25868810/207890783-5f706b1c-4ae6-402f-890e-525284b5cab5.png)
CiliumPod memory usage around restart time. ~ 12:30

**Comment by pchaigno** [comment]
Are you able to reproduce on the latest v1.12 release? v1.12.0 is fairly old for v1.12.

**Comment by PailaGH** [comment]
we have not tried with latest release of V1.12. do you think any changes done in latest release which will help to resolve?
wrt to above graphs, do you see memory is an issue? It seems to me that we have enough memory.

**Comment by pchaigno** [comment]
> wrt to above graphs, do you see memory is an issue? It seems to me that we have enough memory.

Yes, it looks like you still have memory at all times. I still wouldn't recommend to define resource limits in Cilium pods though.

> we have not tried with latest release of V1.12. do you think any changes done in latest release which will help to resolve?

I don't have anything in mind, but in general, the .0 releases tend to have more bugs than the .1, .2, etc. releases. So it's probably worth a try if you want to fix this.

**Comment by PailaGH** [comment]
when I see the reason for termination says Error not OOMKilled
    Last State:     Terminated
      Reason:       Error
      Exit Code:    137
      Started:      Tue, 13 Dec 2022 10:32:22 -0500
      Finished:     Tue, 13 Dec 2022 12:37:22 -0500

what other reason it could be?

**Comment by PailaGH** [comment]
There are ~8 application pods running on the same worker node where the cilium pod is running. these application pods resource memory request 0.5GB per pod and limit is about 2GB per pod. The worker node has total memory of 32GB which means that the cilium pod should be able to use remaining (32-8*2 = 16GB) but the cilium pod is restarting around ~200MB? which is confusing because according to the memory allocations.

as per above cilium memory consumption graph, what could cause the spike in memory consumption?

**Comment by PailaGH** [comment]
I do see following in the logs:  

2022-12-13T12:01:45.323934484Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/cilium_network_policy.go:155: Failed to watch *v2.CiliumNetworkPolicy: the server has asked for the client to provide credentials (get ciliumnetworkpolicies.cilium.io)" subsys=k8s
2022-12-13T12:01:45.323950242Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/service.go:71: Failed to watch *v1.Service: the server has asked for the client to provide credentials (get services)" subsys=k8s
2022-12-13T12:01:45.323953960Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/node.go:106: Failed to watch *v1.Node: the server has asked for the client to provide credentials (get nodes)" subsys=k8s
2022-12-13T12:01:45.323994503Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/namespace.go:65: Failed to watch *v1.Namespace: the server has asked for the client to provide credentials (get namespaces)" subsys=k8s
2022-12-13T12:01:45.324010849Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/pod.go:138: Failed to watch *v1.Pod: the server has asked for the client to provide credentials (get pods)" subsys=k8s
2022-12-13T12:01:45.324025153Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/cilium_clusterwide_network_policy.go:95: Failed to watch *v2.CiliumClusterwideNetworkPolicy: the server has asked for the client to provide credentials (get ciliumclusterwidenetworkpolicies.cilium.io)" subsys=k8s
2022-12-13T12:01:45.324146914Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/network_policy.go:73: Failed to watch *v1.NetworkPolicy: the server has asked for the client to provide credentials (get networkpolicies.networking.k8s.io)" subsys=k8s
2022-12-13T12:01:46.147893877Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/endpoint_slice.go:156: Failed to watch *v1.EndpointSlice: the server has asked for the client to provide credentials (get endpointslices.discovery.k8s.io)" subsys=k8s
2022-12-13T12:01:46.236819889Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/pod.go:138: failed to list *v1.Pod: Unauthorized" subsys=klog
2022-12-13T12:01:46.236838646Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/pod.go:138: Failed to watch *v1.Pod: failed to list *v1.Pod: Unauthorized" subsys=k8s
2022-12-13T12:01:46.497651416Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/node.go:106: failed to list *v1.Node: Unauthorized" subsys=klog
2022-12-13T12:01:46.497675504Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/node.go:106: Failed to watch *v1.Node: failed to list *v1.Node: Unauthorized" subsys=k8s
2022-12-13T12:01:46.527210439Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/cilium_clusterwide_network_policy.go:95: failed to list *v2.CiliumClusterwideNetworkPolicy: Unauthorized" subsys=klog
2022-12-13T12:01:46.527227544Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/cilium_clusterwide_network_policy.go:95: Failed to watch *v2.CiliumClusterwideNetworkPolicy: failed to list *v2.CiliumClusterwideNetworkPolicy: Unauthorized" subsys=k8s
2022-12-13T12:01:46.547012359Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/cilium_endpoint.go:91: Failed to watch *v2.CiliumEndpoint: the server has asked for the client to provide credentials (get ciliumendpoints.cilium.io)" subsys=k8s
2022-12-13T12:01:46.626995915Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/namespace.go:65: failed to list *v1.Namespace: Unauthorized" subsys=klog
2022-12-13T12:01:46.627014311Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/namespace.go:65: Failed to watch *v1.Namespace: failed to list *v1.Namespace: Unauthorized" subsys=k8s
2022-12-13T12:01:46.683613354Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/service.go:71: failed to list *v1.Service: Unauthorized" subsys=klog
2022-12-13T12:01:46.683635044Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/service.go:71: Failed to watch *v1.Service: failed to list *v1.Service: Unauthorized" subsys=k8s
2022-12-13T12:01:46.736323862Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/cilium_network_policy.go:155: failed to list *v2.CiliumNetworkPolicy: Unauthorized" subsys=klog
2022-12-13T12:01:46.736335770Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/cilium_network_policy.go:155: Failed to watch *v2.CiliumNetworkPolicy: failed to list *v2.CiliumNetworkPolicy: Unauthorized" subsys=k8s
2022-12-13T12:01:46.837938768Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/cilium_node.go:130: Failed to watch *v2.CiliumNode: the server has asked for the client to provide credentials (get ciliumnodes.cilium.io)" subsys=k8s
2022-12-13T12:01:46.837951273Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/network_policy.go:73: failed to list *v1.NetworkPolicy: Unauthorized" subsys=klog
2022-12-13T12:01:46.837954218Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/network_policy.go:73: Failed to watch *v1.NetworkPolicy: failed to list *v1.NetworkPolicy: Unauthorized" subsys=k8s
2022-12-13T12:01:46.899585307Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/identitybackend/identity.go:365: Failed to watch *v2.CiliumIdentity: the server has asked for the client to provide credentials (get ciliumidentities.cilium.io)" subsys=k8s
2022-12-13T12:01:47.151834248Z level=warning msg="github.com/cilium/cilium/pkg/k8s/watchers/endpoint_slice.go:156: failed to list *v1.EndpointSlice: Get \"https://172.20.0.1:443/apis/discovery.k8s.io/v1/endpointslices?resourceVersion=9314763\": dial tcp 172.20.0.1:443: connect: connection refused" subsys=klog
2022-12-13T12:01:47.151855033Z level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/watchers/endpoint_slice.go:156: Failed to watch *v1.EndpointSlice: failed to list *v1.EndpointSlice: Get \"https://172.20.0.1:443/apis/discovery.k8s.io/v1/endpointslices?resourceVersion=9314763\": dial tcp 172.20.0.1:443: connect: connection refused" subsys=k8s
2022-12-13T12:01:49.910603610Z level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:app\":\"insights-admin\",\"k8s:io.kubernetes.pod.namespace\":\"ns-prodcac1\"}} NodeSelector:{} Ingress:[{IngressCommonRule:{FromEndpoints:[{\"matchLabels\":{\"k8s:app\":\"k8s-ingress\",\"k8s:io.kubernetes.pod.namespace\":\"ns-prodcac1\"}}] FromRequires:[] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} ToPorts:[{Ports:[{Port:8080 Protocol:TCP}] TerminatingTLS:<nil> OriginatingTLS:<nil> Rules:<nil>}] ICMPs:[]}] IngressDeny:[] Egress:[] EgressDeny:[] Labels:[k8s:io.cilium.k8s.policy.derived-from=NetworkPolicy k8s:io.cilium.k8s.policy.name=insights-admin-ingress-network-policy k8s:io.cilium.k8s.policy.namespace=ns-prodcac1 k8s:io.cilium.k8s.policy.uid=8ceb7ee9-494e-4a03-89c7-95fdfd0186fe] Description:}]" policyAddRequest=ed86be9f-d5c6-47c4-b1ef-2661ea5e0322 subsys=daemon

**Comment by christarazi** [comment]
@PailaGH There seems to be an issue with the K8s apiserver. Cilium seems to be retrying its connections, which is unusual and typically a sign that the apiserver is under load. Is that the case? Are you on managed Kuberentes or is this your own K8s deployment? 

If you can show `cilium_k8s_client_api_latency_time_seconds` metric alongside the memory graph, then we could correlate these two things together to prove / disprove this theory.

**Comment by PailaGH** [comment]
we are using managed Kubernetes. Looking at the information you have asked.

**Comment by PailaGH** [comment]
we did confirm EKS API server restarts at the same as Cilium pod restart.
Is this issue related to 
https://github.com/cilium/cilium/issues/20915

**Comment by christarazi** [comment]
It seems so, yes.

**Comment by PailaGH** [comment]
Is upgrading to 1.12.4 or 1.12.5 a solution?
[arusahni](https://github.com/arusahni) mentioned that he did not observe in 1.12.5

**Comment by PailaGH** [comment]
we have upgraded to 1.12.4 and did not observe the issue so far (10days).

**Comment by pchaigno** [comment]
Good to know! Please comment here if it happens again with a recent release.
